### PR TITLE
React - Stepper - Add possibility for label to be a ReactNode

### PR DESCRIPTION
**Pull request by ofreyssinet-ledger**
On LLD the Stepper implementation currently accepts a ReactNode for the `label` prop.
This non breaking change makes it easier to use this lib's Stepper in LLD.

### CODE REVIEW

File: packages/react/src/components/navigation/progress/Stepper/index.tsx [hunk 1]
```diff
@@ -151,9 +152,13 @@ export const Step = memo(function Step({
...
       </Item.Spacer>
-      <StepText inactive={inactive} errored={errored} variant="small">
-        {label}
-      </StepText>
+      {typeof label === "string" ? (
+        <StepText inactive={inactive} errored={errored} variant="small">
+          {label}
+        </StepText>
+      ) : (
+        label
```
Comment (elbywan) [inline packages/react/src/components/navigation/progress/Stepper/index.tsx:160]: The issue with passing a plain react node as a label is that it would be impossible to style depending on the step and its state (active/inactive/errored).

Comment (ofreyssinet-ledger) [reply packages/react/src/components/navigation/progress/Stepper/index.tsx:160]: indeed, I'll change this then!

File: packages/react/src/components/navigation/progress/Stepper/index.tsx [hunk 2]
```diff
@@ -133,13 +143,12 @@ const stepContentsByState = {
 
 export const Step = memo(function Step({
   state,
-  label,
-  hideLeftSeparator,
+  label: Label,
+  hideLeftSeparator = true,
```
Comment (elbywan) [inline packages/react/src/components/navigation/progress/Stepper/index.tsx:147]: Wondering why this change (defaults to `true`) here?

File: packages/react/src/components/navigation/progress/Stepper/index.tsx [hunk 3]
```diff
@@ -1,17 +1,28 @@
...
-type Label = string | ReactNode;
+/**
+ * The state of a progress bar step.
+ */
+export type StepState = "pending" | "current" | "completed" | "errored";
+
+type LabelType =
+  | string
+  | React.ComponentType<{ state: StepState }>
+  | ((props: { state: StepState }) => JSX.Element);
```
Comment (elbywan) [inline packages/react/src/components/navigation/progress/Stepper/index.tsx:17]: I think that the `React.ComponentType<{ state: StepState }>` type encompass `((props: { state: StepState }) => JSX.Element)`.

**Review by elbywan — APPROVED** [review]
👌

File: packages/react/src/components/navigation/progress/Stepper/index.tsx (hunk 2 above)
Comment (ofreyssinet-ledger) [reply packages/react/src/components/navigation/progress/Stepper/index.tsx:147]: Good catch that's a mistake, will change that
